lib/trino-parquet/src/main/java/io/trino/parquet/ParquetTypeUtils.java

```diff
@@ -129,27 +129,23 @@ public static Optional<ColumnDescriptor> getDescriptor(List<PrimitiveColumnIO> c
 
     private static int getPathIndex(List<PrimitiveColumnIO> columns, List<String> path)
     {
-        int maxLevel = path.size();
-        int index = -1;
         for (int columnIndex = 0; columnIndex < columns.size(); columnIndex++) {
-            ColumnIO[] fields = columns.get(columnIndex).getPath();
-            if (fields.length <= maxLevel) {
+            String[] fieldPath = columns.get(columnIndex).getFieldPath();
+            if (fieldPath.length != path.size()) {
```

Comment:
Can you pls spend some extra time to see whether `io.trino.parquet.ParquetTypeUtils#getDescriptors` performance can be improved . I don't have any hints on where to dig/what to change at the moment :(

Reply:
That's needed because the `descriptorsByPath` map has List<String> as the key, switching to String[] won't work because the equals method is shallow comparision.